cmd/juju/model/destroy.go

```diff
@@ -101,18 +101,33 @@ const destroyExamples = `
 `
 
 var destroyModelMsg = `
-This command will destroy the %q model and all its resources. It cannot be stopped.`[1:]
+This command will destroy the %q model and affect the following resources. It cannot be stopped.`[1:]
 
 var destroyModelMsgDetails = `
 {{- if gt .MachineCount 0}}
- - {{.MachineCount}} {{if .IsCaaS}}container{{else}}machine{{end}}{{if gt .MachineCount 1}}s{{end}} will be destroyed
-  - {{if .IsCaaS}}container{{else}}machine{{end}} list:{{range .MachineIds}} "{{.}}"{{end}}
+ - {{.MachineCount}} machine{{if gt .MachineCount 1}}s{{end}} will be destroyed
+  - machine list:{{range .MachineIds}} "{{.}}"{{end}}
+{{- end}}
  - {{.ApplicationCount}} application{{if gt .ApplicationCount 1}}s{{end}} will be removed
  {{- if gt (len .ApplicationNames) 0}}
   - application list:{{range .ApplicationNames}} "{{.}}"{{end}}
  {{- end}}
  - {{.FilesystemCount}} filesystem{{if gt .FilesystemCount 1}}s{{end}} and {{.VolumeCount}} volume{{if gt .VolumeCount 1}}s{{end}} will be {{if .ReleaseStorage}}released{{else}}destroyed{{end}}
-{{- end}}
+`
+
+var persistentStorageErrorMsg = `cannot destroy model "{{.ModelName}}"
+
+The model has persistent storage remaining:
+	{{.DetachableVolumeCount}} volume{{if gt .DetachableVolumeCount 1}}s{{end}} and {{.DetachableFilesystemCount}} filesystem{{if gt .DetachableFilesystemCount 1}}s{{end}}
```

Comment:
Can we improve this - if vol or fs is 0, don't print it.

Reply:
Fixed.  go template go looks real ugly, but works.  If there's a better way though I'm open to it

Reply:
Left a suggestion in a comment - just use Sprintf